packages/implementations-lib/src/transfers/mongo_transfers_repo.ts

```diff
@@ -97,153 +98,189 @@ export class MongoTransfersRepo implements ITransfersRepository {
-                throw new UnableToGetTransferError();
-            });
-
-            const countResult = await this.transfers.countDocuments(filter).catch(reason => {
-                this._logger.error("Unable to get transfers count");
-            }) || result.length;
-
-            searchResults.items = result.map(this._mapToTransfer);
-
-            searchResults.totalPages = Math.ceil(countResult / pageSize);
-            searchResults.pageSize = Math.max(pageSize, result.length);
-
-        } catch (err) {
-            this._logger.error(err);
-        }
-
-        return Promise.resolve(searchResults);
+		payeeDfspName: string | null,
+		payerDfspName: string | null,
+		state: string | null,
+		transferType: string | null,
+		payerIdType: string | null,
+		payeeIdType: string | null,
+		currency: string | null,
+		id: string | null,
+		payerIdValue: string | null,
+		payeeIdValue: string | null,
+		startDate: number,
+		endDate: number,
+		pageIndex = 0,
+		pageSize: number = MAX_ENTRIES_PER_PAGE
+	): Promise<TransfersSearchResults> {
+		// make sure we don't go over or below the limits
+		pageSize = Math.min(pageSize, MAX_ENTRIES_PER_PAGE);
+		pageIndex = Math.max(pageIndex, 0);
+
+		const searchResults: TransfersSearchResults = {
+			pageSize: pageSize,
+			pageIndex: pageIndex,
+			totalPages: 0,
+			items: []
+		};
+
+		let filter: any = { $and: [] }; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+		if (id) {
+			filter.$and.push({ "transferId": { "$regex": id, "$options": "i" } });
+		}
+		if (state) {
+			filter.$and.push({ transferState: state });
+		}
+		if (currency) {
+			filter.$and.push({ currencyCode: currency });
+		}
+		if (startDate) {
+			filter.$and.push({ updatedAt: { $gte: startDate } });
+		}
+		if (endDate) {
+			filter.$and.push({ updatedAt: { $lte: endDate } });
+		}
+
+		if (payerDfspName) {
+			filter.$and.push({ payerFspId: payerDfspName });
+		}
+
+		if (payeeDfspName) {
+			filter.$and.push({ payeeFspId: payeeDfspName });
+		}
+
+		if (payerIdValue) {
+			filter.$and.push({ payerFspId: payerIdValue });
+		}
+
+		if (payeeIdValue) {
+			filter.$and.push({ payeeFspId: payeeIdValue });
+		}
+
+		if (filter.$and.length === 0) {
+			filter = {};
+		}
+
+		try {
+			const totalRecordsCount = await this.transfers.countDocuments(filter);
+			const results = await this.transfers.find(
+				filter,
+				{
+					sort: ["updatedAt", "desc"],
+					projection: { _id: 0 },
+					skip: Math.floor(pageIndex * pageSize),
+					limit: pageSize
+				}
+			).toArray().catch((e: unknown) => {
+				this._logger.error(`Unable to get transfers: ${(e as Error).message}`);
+				throw new UnableToGetTransferError();
+			});
+
+			if (transferType || payerIdType || payeeIdType) {//Perform deep search by decoding ILP packet if these search params included
+
+				const filteredResults = results.filter((result) => {
+					const decodedIlpPacketData = this._decodeIlpPacket(result.ilpPacket) as any;
```

Comment:
We shouldn't need to decode this here, should be in the AdminUI, see transfers detail and quotes detail pages, if we need these these fields in the transfer object, we need to put them in, not do it like this. 

@ruirochawork can we add scenario.transfeType, payer and payee partyTypes?

Reply:
Got it @pedrosousabarreto, I just saw this comment.